api_service.py

```
@@ -2836,6 +2857,30 @@ async def not_found(request, exc):
+    logger.error(f"Request validation error on {request.method} {request.url.path}")
+    logger.error(f"Validation errors: {exc.errors()}")
+
+    # Log request details for OAuth callback specifically
+    if "/api/auth/gmail/callback" in str(request.url.path):
+        logger.error(f"OAuth callback validation failed - URL: {request.url}")
+        logger.error(f"OAuth callback query params: {dict(request.query_params)}")
+        try:
+            body = await request.body()
+            logger.error(f"OAuth callback request body: {body.decode('utf-8', errors='replace')[:500]}")
+        except Exception as e:
+            logger.exception(f"Could not read request body: {e}")
```

Ruff (0.14.10)

Redundant exception object included in `logging.exception` call

(TRY401)